playbook/workshops/remote_control/README.md

```diff
@@ -0,0 +1,615 @@
+
+> ![](img/hiding_and_showing_sections.gif)
+
+Before you do anything, in the JavaScript tab, change the `555-555-5555` phone
+number to your number.
+
+> ![](img/change_phone_number.gif)
+
+Then click the checkbox for "Auto-run JS"
+
+> ![](img/auto_run_js.gif)
+
+When you click the button that says "Emergency Excuse", it will call your phone
+and say "Hello! Are you ready for MHacks?" (Ths feature might come in handy when
+you are on an awkward date and you need your phone to ring...)
+
+Go ahead and click the "Emergency Excuse" button:
+
+> ![](img/button_press.gif)
+
+> If it doesn't, ask a neighboring group to see if they can see what's wrong.
+> Otherwise, raise your hand to ask a facilitator for help!
+
+#### Trying to understand the code.
+
+There are two pieces of code that make this work, the code in the HTML:
+
+```html
+<button id="emergency-button">Emergency Excuse</button>
+```
+
+located here:
+
+> ![](img/button.png)
+
+And the code in the JavaScript:
+
+```js
+document.getElementById("emergency-button").onclick = function() {
+  Twilio.callAndSay("555-555-5555", "This is your boss! Get over here!");
+  alert("Calling...");
+};
+```
+
+With your parter (and maybe some neighbors), spend just 1-2 minutes trying to
+talk through how you think this code works. (Hint: say the JavaScript out loud
+**slowly**...)
+
+#### Understanding the code
+
+Instead of explaining how the existing code works, I will explain by going
+through the process of adding another button. When I press this new button, it
+will call me and say "Time to wake up! Keep hacking! Just do it!".
+
+##### Adding the button
+
+In the HTML let's add another button by writing:
+
+```html
+<button>Motivate Partner</button>
+```
+
+> ![](img/motivate_partner.gif)
+
+<!--Result-->
+
+##### Adding an id to the button
+
+Then we need to give this button some id so we can refer to it by its id in the
+JavaScript.
+
+```html
+<button id="motivation-button">Motivate Partner</button>
+```
+
+> ![](img/motivate_partner_2.gif)
+
+##### Referencing the button from JavaScript
+
+Now that we set the button's id, we add this to our JavaScript:
+
+```js
+document.getElementById("motivation-button").onclick = function() {
+    [PUT ANY CODE HERE YOU WANT WHEN THE BUTTON IS CLICKED]
+};
+```
+_Don't include the `[PUT ANY CODE HERE YOU WANT WHEN THE BUTTON IS CLICKED]`_
+
+> ![](img/getElement.gif)
+
+There is a lot to understand in the above code but if you read it out loud like
+this, it might begin to make sense:
+
+_Note that I highlighted the keywords from the code_
+
+In the `document`, `get` the `element` with the `id` `"motivation-button"` and
+when it is `clicked`, run any of the code that is inside the `function`.
+
+There are definitely more concepts and complexities here but we won't worry
+about them in this workshop. If you want to read more about what's happening
+here, feel free to Google `javascript function` and `document.getElementById`
```

Comment:
totally link to: http://lmgtfy.com/?q=javascript+function and http://lmgtfy.com/?q=document.getElementById :laughing: